Source: https://gitlab.gnome.org/GNOME/empathy/-/merge_requests/3
Upstream: no, because upstream is completely abandoned

From f42c415154c797a1b3fbddde60ad8fedb2eda988 Mon Sep 17 00:00:00 2001
From: Laurent Bigonville <bigon@bigon.be>
Date: Tue, 7 Jan 2020 11:09:26 +0100
Subject: [PATCH 1/2] libempathy-gtk: Stop using deprecated enchant function

Replace enchant_dict_add_to_pwl() by enchant_dict_add() and bump minimal
version accordingly
---
 configure.ac                   | 2 +-
 libempathy-gtk/empathy-spell.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7aeb6b31c..4bf2d4d0c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -72,7 +72,7 @@ WEBKIT_REQUIRED=2.10.0
 GOA_REQUIRED=3.5.1
 
 # Optional deps
-ENCHANT_REQUIRED=1.2.0
+ENCHANT_REQUIRED=1.4.0
 GEOCLUE_REQUIRED=2.1.0
 GEOCODE_GLIB_REQUIRED=0.99.1
 ISO_CODES_REQUIRED=0.35
diff --git a/libempathy-gtk/empathy-spell.c b/libempathy-gtk/empathy-spell.c
index 27f27f902..bb0159fd8 100644
--- a/libempathy-gtk/empathy-spell.c
+++ b/libempathy-gtk/empathy-spell.c
@@ -424,7 +424,7 @@ empathy_spell_add_to_dictionary (const gchar *code,
 	if (lang == NULL)
 		return;
 
-	enchant_dict_add_to_pwl (lang->speller, word, strlen (word));
+	enchant_dict_add (lang->speller, word, strlen (word));
 }
 
 #else /* not HAVE_ENCHANT */
-- 
GitLab


From 6c12a411274acf86e494b36704abb8e18b56642e Mon Sep 17 00:00:00 2001
From: Laurent Bigonville <bigon@bigon.be>
Date: Tue, 7 Jan 2020 11:38:13 +0100
Subject: [PATCH 2/2] configure.ac: Switch to enchant-2

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4bf2d4d0c..14f83c852 100644
--- a/configure.ac
+++ b/configure.ac
@@ -72,7 +72,7 @@ WEBKIT_REQUIRED=2.10.0
 GOA_REQUIRED=3.5.1
 
 # Optional deps
-ENCHANT_REQUIRED=1.4.0
+ENCHANT_REQUIRED=2.1.1
 GEOCLUE_REQUIRED=2.1.0
 GEOCODE_GLIB_REQUIRED=0.99.1
 ISO_CODES_REQUIRED=0.35
@@ -307,7 +307,7 @@ AC_ARG_ENABLE(spell,
 if test "x$enable_spell" != "xno"; then
    PKG_CHECK_MODULES(ENCHANT,
    [
-      enchant >= $ENCHANT_REQUIRED,
+      enchant-2 >= $ENCHANT_REQUIRED,
       iso-codes >= $ISO_CODES_REQUIRED
    ], have_enchant="yes", have_enchant="no")
 
-- 
GitLab

